### DISCUSSION TITLE
Your login session has expired

### DISCUSSION THREAD

**Discussion by 0PandaDEV** [dani-garcia/vaultwarden#4891] [Q&A] [answered]
### Vaultwarden Support String

For some reason I cannot get my support string, it shows Error generating

### Vaultwarden Build Version

 1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

cloudflare tunnels zero trust

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 23.10

### Clients

Browser Extension, iOS

### Client Version

Arc 1.58.0 - v2024.7.1, IOS 18 - v2024.7.1

### Steps To Reproduce

Probably not reproducible:
1. try logging into the instance from IOS or the browser extension
2. it will fail

### Expected Result

Normal usage/login

### Actual Result

I get an error like this, "Your login session has expired" and then get sent back to the login screen.

### Logs

```text
[2024-08-24 01:06:32.535][request][INFO] GET /api/devices/knowndevice
[2024-08-24 01:06:32.537][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
[2024-08-24 01:06:39.663][request][INFO] POST /identity/accounts/prelogin
[2024-08-24 01:06:39.663][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
[2024-08-24 01:06:40.605][request][INFO] POST /identity/connect/token
[2024-08-24 01:06:40.992][vaultwarden::api::identity][INFO] User ***REDACTED FOR PRIVACY logged in successfully. IP: 172.17.0.1
[2024-08-24 01:06:40.992][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-08-24 01:06:41.367][request][INFO] POST /identity/connect/token
[2024-08-24 01:06:41.373][response][INFO] (login) POST /identity/connect/token => 200 OK
```

### Screenshots or Videos

https://github.com/user-attachments/assets/c42f4240-31dc-410b-9e84-6817174d2351

### Additional Context

This issue started occurring on 23 August 2024.

**Comment by BlackDex** [comment]
The login in successful according to the logs. 
Not being able to generate the support string is strange. 

But if you get your session is expired i would suggest to check the date and time on your client and server. 
This should also be visible on the diagnostics page it self without generating the support string.

**Reply by 0PandaDEV** [reply]
Ok because my instance is in Switzerland, but I'm currently in Australia could that also be a problem?

**Reply by BlackDex** [reply]
If your computer is still on European time then yes

**Reply by BlackDex** [reply]
Or somehow manually changed it so the timezone is still Switzerland but you changed the time to match Australia.

**Reply by 0PandaDEV** [reply]
Both my Mac and my iPhone are set to use the Australian time and region, so this should technically work, and it also did work for a few days when I was in Australia, but then it suddenly stopped working.

**Reply by BlackDex** [reply]
Check the time on your devices. It might be out of sync. 
https://time.is/

But this could be on your server too.

**Reply by 0PandaDEV** [reply]
How should I check this on my server, it's headless

**Reply by BlackDex** [reply]
Maybe this helps. 
https://www.cyberciti.biz/faq/linux-unix-bsd-is-ntp-client-working/

**Reply by 0PandaDEV** [reply]
I changed my instance to docker compose and now the admin panel finally works properly, and I can generate a support string, but the login on all clients is still bugged.

What I also found out was that NTP is not working for some reason.

```bash
$ timedatectl             
			   Local time: Sun 2024-08-25 01:39:12 CEST
           Universal time: Sat 2024-08-24 23:39:12 UTC
                 RTC time: Sat 2024-08-24 23:39:12
                Time zone: Europe/Zurich (CEST, +0200)
System clock synchronized: yes
              NTP service: n/a
          RTC in local TZ: no
```

```bash
$ timedatectl set-ntp true
Failed to set ntp: NTP not supported
```

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: false
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**********************",
  "domain_origin": "*****://**********************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by 0PandaDEV** [reply]
Never mind saving, the admin panel gives me this again 
![image](https://github.com/user-attachments/assets/b56f99ea-f6c4-4354-95d9-dda216e687cb)

**Reply by BlackDex** [reply]
Strange. There must be something with either a reverse proxy, or something in between your client and the Vaultwarden server. 
Can't really think of something else right now.

**Reply by 0PandaDEV** [reply]
OMG, I just noticed I forgot to put https:// before my custom instance URL, that was the problem.
But I could not find a pattern I think the problem is the Cloudflare tunnel, but that one also sometimes works, so idk what is going on.

**Reply by 0PandaDEV** [reply, marked as answer]
Ok fixed it this time the problem was WAF from Cloudflare blocked some Bitwarden API requests.
